src/legacy/core_plugins/kibana/public/visualize/plugin.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { SavedVisualizations } from './saved_visualizations/saved_visualizations';
+
+import { Dependencies, Management, Ui, Visualize } from '../../plugin_dependencies';
+
+// TODO: import real type from core once available
+interface CoreStart {
+  [key: string]: any;
+}
+
+export interface VisualizeStartDependencies extends Dependencies {
+  management: Pick<Management, 'savedObjectManagementRegistry'>;
+  ui: Pick<
+    Ui,
+    | 'createLegacyClass'
+    | 'savedObjectLoader'
+    | 'savedObjectProvider'
+    | 'savedObjectsClientProvider'
+    | 'uiModules'
+  >;
+  visualize: Pick<Visualize, 'updateOldState' | 'visProvider' | 'visTypesRegistryProvider'>;
+}
```

Comment:
This approach using `Pick` saves you from rewriting types, but still feels tedious. I'd prefer to not extend `Dependencies` since they are intended to be temporary, but I suppose we'll need to update types here down the line anyway, as proper types begin to be exposed from other plugins and can then be imported here.

Regardless, would welcome any better ideas 😃